### PR TITLE
10/get/api/articles queries

### PR DESCRIPTION
**Pull request by majid-L**

### CODE REVIEW

File: models/utility-queries.models.js [hunk 1]
```diff
@@ -1,10 +1,34 @@
...
       }
     });
 };
+
+exports.checkTopicExists = topic =>{
+  if (topic) {
+    return db.query(`SELECT slug FROM topics WHERE slug = $1;`, [topic])
+    .then(({rowCount}) => {
+      if(rowCount === 0) {
+        return Promise.reject({status: 404, msg : 'Topic not found.'});
```
Comment (popatre) [inline models/utility-queries.models.js:18]: This says topic not found, but the test expects `slug not found`

Comment (popatre) [reply models/utility-queries.models.js:18]: I think this isnt actually used anywhere

**Review by popatre — CHANGES_REQUESTED** [review]

File: __tests__/app.test.js [hunk 2]
```diff
@@ -346,5 +346,86 @@ describe("9) GET /api/articles/:article_id (comment count)", () => {
...
+    expect(msg).toBe('slug not found.');
+  })
+})
+
+it("Responds with 200 status code and an empty array if client's request includes a topic query for a topic that exists in the database but which is not referenced by any of the article objects.", () => {
+  return request(app)
+  .get('/api/articles?topic=paper')
+  .expect(200)
+  .then(({ body : { articles }}) => {
+    expect(articles).toEqual([]);
```
[the remark this answers is not in the document]
Comment (3782291211) [reply __tests__/app.test.js:446]: I've added these in now.

File: controllers/news.controllers.errors.js [hunk 3]
```diff
@@ -7,11 +7,12 @@ exports.handlePSQLerrors = (err, req, res,next) =>
   : next(err);
 
 exports.handleForeignKeyError = (err, req, res,next) => {
-  err.code === '23503' ? 
-    err.detail.includes('article') ? res.status(404).send({msg: "Article not found."})
-    : err.detail.includes('author') ? res.status(404).send({msg: "Username not found."})
-    : res.status(404).send({msg: "Resource not found."})
-  : next(err);
+  if (err.code === '23503') {
+     const columnName = err.detail.match(/\(([a-zA-Z]+)/)[1];
```
[the remark this answers is not in the document]
Comment (3782291211) [reply controllers/news.controllers.errors.js]: I've re-worked the error handler, but I'm still not 100% sure it's quite right. I'm happy to make it more generic and just have it send "Resource not found" for all the errors that it deals with.

File: models/news.models.js [hunk 4]
```diff
@@ -5,15 +5,30 @@ exports.selectTopics = () => {
   .then(({rows : topics}) => topics);
 };
 
-exports.selectArticles = () => {
-  return db.query(`
-  SELECT articles.author, title, articles.article_id, topic, articles.created_at, articles.votes, COUNT(comment_id) ::INTEGER AS comment_count
+exports.selectArticles = queryObj => {  
+  const sort_by = queryObj.sort_by ? queryObj.sort_by : 'created_at';
```
[the remark this answers is not in the document]
Comment (3782291211) [reply models/news.models.js]: Done! Great suggestion. The code looks much cleaner in that section now.

File: models/utility-queries.models.js (hunk 1 above)
Comment (3782291211) [reply models/utility-queries.models.js]: Forgot to delete this function!

**Review by Poonam-raj — APPROVED** [review]
Excellent stuff - feel free to merge!

**Pull request merged**